src/main/java/soot/FastHierarchy.java

```diff
@@ -763,10 +770,21 @@ private SootMethod resolveMethod(
       Set<SootClass> ignoreList) {
     SootClass concreteType = baseType;
 
+    
+    SootMethod candidate = null;
```

Comment:
Generate a signature with soot.SootMethod#getSubSignature(java.lang.String, java.util.List<soot.Type>, soot.Type) as key for the inner map of vtable

Reply:
Use the declaring class as key for the outer map